enhancements/cluster-validation-tool.md

```diff
@@ -0,0 +1,521 @@
+expected user variations. The hugepage node is indicated as extra content and the realtime kernel
+setting is indicated as a drift
+
+```diff
+---
+apiVersion: performance.openshift.io/v2                          apiVersion: performance.openshift.io/v2
+kind: PerformanceProfile                                         kind: PerformanceProfile
+metadata:                                                        metadata:
+  annotations:                                                     annotations:
+    ran.openshift.io/ztp-deploy-wave: "10"                           ran.openshift.io/ztp-deploy-wave: "10"
+  name: openshift-node-performance-profile                         name: openshift-node-performance-profile
+spec:                                                            spec:
+  additionalKernelArgs:                                            additionalKernelArgs:
+    - rcupdate.rcu_normal_after_boot=0                               - rcupdate.rcu_normal_after_boot=0
+  cpu:                                                             cpu:
+    isolated: 2-19,22-39                                             isolated: 2-31,34-63
+    reserved: 0-1,20-21                                              reserved: 0-1,32-33
+  hugepages:                                                       hugepages:
+    defaultHugepagesSize: 1G                                         defaultHugepagesSize: 1G
+    pages:                                                           pages:
+      - count: 32                                                      - count: 32
+                                                              >          node: 0
+        size: 1G                                                         size: 1G
+  machineConfigPoolSelector:                                       machineConfigPoolSelector:
+    pools.operator.machineconfiguration.openshift.io/master:         pools.operator.machineconfiguration.openshift.io/master:
+  nodeSelector:                                                    nodeSelector:
+    node-role.kubernetes.io/master: ""                               node-role.kubernetes.io/master: ""
+  numa:                                                            numa:
+    topologyPolicy: restricted                                       topologyPolicy: restricted
+  realTimeKernel:                                                  realTimeKernel:
+    enabled: true                                             |      enabled: false
+
+---
+<next CR>
+...
+
+Summary
+  Missing required CRs
+    someCR
+    anotherCR
+  Unmatched CRs
+    unknownCR
+    anotherUnknownCR
+  Total CRs with drift: 2
+```
+
+#### Tool delivery
+
+The tool will be a standalone binary executable suitable for direct use by customers, developers,
+support engineering, etc. The tool may be installed as a kubectl/oc plugin.
+
+### Reference Configuration Specification
+
+The reference configuration consists of a set of CRs (yaml files with one CR per file) with
```

Comment:
Will be somehow encode the version of the reference config to the configuration itself? Just so we can always be sure what we got even if it is just a zip file from somewhere? Relying purely on git branches is probably cause a lot of mess.

In other words, I am looking for a file like /etc/redhat-release is in RHEL.

Reply:
Yes, I think this would be quite valuable. It would likely be helpful to have a consistent "entry point" into the reference configuration which has a version and perhaps other metadata (eg reference configuration name).
Something like:

referenceConfig.yaml
```
name: "Some Reference Config"
version: x.y.z
```